### ISSUE TITLE
[Communication] Prepare progress communication for Kepler's team 

### ISSUE THREAD

**Issue by ana-md** [CartoDB/kepler.gl#25]
## Background

After a couple of conversations with Victor and Mireya, we realised that all the work we have been doing so far for Kepler's integration should be understood as only **ONE PRODUCT FEATURE** with different possibilities. 

We decided to call it using the general term **Backend Storage** (just until we align with Virginia for a proper product naming). This way, we use internally just this naming to avoid confusion, as it's the way Kepler also refers to it.

This is a feature that includes all the work we have been dividing so far into different "features": share map via URL (v0.1), Storage Engine (v0.2),... However, instead of dividing it into different features, they will all be considered from now on as **Development Phases** from just ONE PRODUCT FEATURE (**Backend Storage**), to make our work easier to coordinate with Kepler. 

These Development Phases will all have an issue + PR associated in Kepler's repo:

**1. Share public MAP** (via URL) - v0.1
   - Issue 1 has already been created in Kepler's repo (issue 746)
   - [PR (v0.1)](https://github.com/CartoDB/kepler.gl/issues/21) is the main goal for this sprint (once the blockers with the Signup and OAuth are solved) 

**2. Persist MAPS (privately)** (part of former "Storage Engine") - v0.2
   - Issue 2 draft will be created here for review before sharing it in Kepler's repo. 
   - PR (v0.2) 

**3. Persist DATA (privately)** (part of former "Storage Engine") - v0.3
   - Issue 3 (more specific details about datasets)
   - PR (v0.3)

## Goal

The **goal for this week** is to collect all the info for progress communication for Kepler's PM (Shan) regarding the status of the different development phases mentioned above:

**1. Share public MAP** (via URL) - v0.1
As we can't create the PR (v0.1) until the blockers from backend are solved and the v0.1 properly tested afterwards, we will share a [demo video](https://github.com/CartoDB/kepler.gl/issues/28) with them (without OAuth) and let them know PR1 will be there soon. 

**2. Persist MAPS** (part of former "Storage Engine") - v0.2
We will create Issue 2 in their repo and share with them the link with the [latest prototype](https://projects.invisionapp.com/share/TPTSVDX2KF5#/screens/389126759_Storage_V3) that we all have reviewed. 
_Note: If we have time to get the mockups for the [CARTO Dashboard ](https://github.com/CartoDB/product/issues/467)ready and reviewed, we will also include them in the prototype that we share with them._ 

**3. Persist DATA (privately)** (part of former "Storage Engine") - v0.3
Let them know we are working on the definition and design of this last phase as it has higher challenges due to the way that both Kepler and CARTO work differently with datasets. 
Besides, ping them to the 2 issues (blockers) that Román has already created in their repo. 

Once all of this is ready, I will collect all the information and share it in an e-mail with her.

**Comment by ana-md** [comment]
## Backend Storage using CARTO

**Is your feature request related to a problem? Please describe.**
Kepler.gl has a 'Save/Share your map' feature using Dropbox but it doesn't have a way to privately manage your maps and data after its creation, something that can be frustrating after spending hours working on the creation of your maps.  

**Describe the solution you'd like**
Having an integrated and private system to easily persist, load and manage your maps and data from an external provider. 

**Describe alternatives you've considered**
We propose a generic approach, in which any cloud provider with an OAuth API can include a predefined set of methods to implement the backend storage feature. This feature will allow the user to persist maps and datasets, with privacy control, and to get them back from the provider. 

As the first backend storage provider, we want to add a set of functionalities that include different possibilities working with your maps and data coming from Kepler (e.g., sharing via URL, saving privately at CARTO, loading back in Kepler, management from your CARTO Dashboard,...).

## Implementation Plan

We propose dividing the implementation of this feature into **3 different development phases** to make team collaboration between CARTO and Kepler easier and faster. 

Each of these development phases will come with both a descriptive issue and a PR associated.
This way, progress for each phase is clearer and can be reviewed in parallel by Kepler’s team. 

Please, keep in mind that **nothing should be merged until all PRs are ready** and it can be publicly released as one single feature (Backend Storage).

The development phases (with their current status) are the following:

1. **Phase 1: Sharing map via URL using CARTO**
  - Issue found here: `Link to already created issue in Kepler's repo`
  - PR for this phase will come soon, implementation is ready from our side (we are just waiting for some final improvements with our OAuth API). 
  - Progress: Find here a demo video that we have prepared for you to take a look: `Link to  demo video`

2. **Phase 2: Backend Storage (I): Persisting MAPS (privately)**
  - Issue found here: `Link to` [new issue](https://github.com/CartoDB/kepler.gl/issues/29) `in Kepler's repo that we will create today`
  - PR not there yet, on-going implementation from our side. 
  - Progress: Find here our latest prototype including all the feedback we already got from you in previous conversations: `Link to InVision Prototype` (without screenshots, those ones will be in the issue directly)
_(Note: if there is any other question/feedback from your side, please share it directly in the issue created for this phase)_

3. **Phase 3: Backend Storage (II): Persisting DATA (privately)**
  - Issue 3 - Under definition (will come soon)
  - PR not there yet
  - Progress: We are working on a proper definition of data management that provides a great user experience, as there are many possibilities and different challenges due to the different approach while working with data that Kepler and CARTO have. 
In the meantime, we have found the following blockers that we have already reported in your repo and would be great if you could take a look at them: 
           - `Link to Issue 791 (Kepler's repo)`
           - `Link to Issue 792 (Kepler's repo)`

**Comment by ana-md** [comment]
Issue created in their repo already!
